.github/workflows/Ci.yml

```diff
@@ -0,0 +1,109 @@
+      run: |
+        source "$INIT_ENV"
+        PACKAGES=$(find . -maxdepth 2 -type f -name package.xml | sed 's/\.\///' | cut -d/ -f1)
+        sed -i 's/colcon build /colcon build --packages-skip novatel_oem7_msgs tracetools tracetools_test /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Run ROS1 C++ Tests
+      # Run ROS1 C++ tests
+      # The skipped ROS 1 packages here should match the ROS 1 packages that are skipped during the build in docker/install.sh"
+      run: |
+        source "$INIT_ENV"
+        sed -i 's/colcon test /colcon test --packages-skip novatel_oem7_msgs tracetools tracetools_test carma_launch_testing carma_message_utilities ros2bag ros2trace tracetools_launch tracetools_read tracetools_trace multiple_object_tracking platoon_control platoon_control_ihp rosbag_mock_drivers /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -t -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Backup ROS1 compile_commands.json
+      # Backup the ROS1 compile_commands.json file for sonarcloud
+      run: |
+        mv /opt/carma/build/compile_commands.json /opt/carma/compile_commands.ros1.json
+
+    - name: Cleanup before ROS 2 build
+      # Clear the build and install folders before building ROS 2
+      run: |
+        rm -rf /opt/carma/install
+        rm -rf /opt/carma/build
+
+    - name: Build ROS2
+      # Build ROS2 packages
+      run: |
+        source "$INIT_ENV"
+        source "$ROS_2_ENV"
+        source /opt/autoware.ai/ros/install_ros2/setup.bash
+        sed -i '/colcon build/ s/$/ --parallel-workers 4 --event-handlers console_direct+ --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_BUILD_TYPE="Debug"/' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Run ROS2 C++ Tests
+      # Run ROS2 C++ tests
+      run: |
+        source "$INIT_ENV"
+        source "$ROS_2_ENV"
+        source /opt/autoware.ai/ros/install_ros2/setup.bash
+        sed -i 's/colcon test /colcon test --packages-skip approaching_emergency_vehicle_plugin basic_autonomy carma_cooperative_perception carma_guidance_plugins carma_wm carma_wm_ctrl cooperative_lanechange emergency_response_vehicle_plugin gnss_to_map_convertor lci_strategic_plugin light_controlled_intersection_tactical_plugin mobilitypath_publisher motion_computation object_visualizer platoon_strategic_ihp pose_to_tf route route_following_plugin sci_strategic_plugin stop_and_dwell_strategic_plugin stop_and_wait_plugin subsystem_controllers system_controller traffic_incident_parser yield_plugin /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash

```

Comment:
Let's remove `-packages-skip` along with the listed packages. This will cause the workflow to fail, but we will address that in separate work (https://usdot-carma.atlassian.net/browse/CAR-6008).

Reply:
removed